dashboard/test/controllers/api/v1/projects/personal_projects_controller_test.rb

```diff
@@ -0,0 +1,37 @@
+require 'test_helper'
+
+class Api::V1::Projects::PersonalProjectsControllerTest < ActionDispatch::IntegrationTest
+  setup do
+    @project_owner = create :user
+
+    personal_project_value = {
+      name: 'Tetris',
+      level: '/projects/gamelab',
+      createdAt: '2017-01-24T16:41:08.000-08:00',
+      updatedAt: '2017-01-25T17:48:12.358-08:00'
+    }.to_json
+    personal_project = {id: 22, value: personal_project_value}
+
+    PEGASUS_DB.stubs(:[]).with(:storage_apps).returns(
+      stub(where: [personal_project])
+    )
```

Comment:
I don't love this set up because I don't have a nice way to test that the logged in user can only see her personal projects.  Is there a better way? Is this worth worrying about?

Reply:
I think this is a great start and we might need to rely on UI tests near the end of this feature to check that.

You could have this test actually insert a database row for the project, check that, and remove it on cleanup; up to you which approach you'd like to pursue.

Reply:
I've had this problem while testing other things in this file. essentially, we are not testing the database query. 

I believe this would become more testable if we had a rails model for projects, and used rails associations to find them, then we could use FactoryBot to create them during tests, and could effectively test that the right thing is happening. Do others have opinions on whether this theory is correct?

I like either of Brad's suggestions for short term solutions.

Reply:
I like this idea, but I don't think it even needs to be an ActiveRecord model with associations necessarily.
Dave, are you suggesting backing it with a DB table? Ideally this would just live in the `dashboard` DB and could be fully contained in Rails. If we have a wrapper model not backed by a dashboard table, that queries the Pegasus DB, and we want to create objects with `FactoryBot` then we'll also need to override `create`, `save`, etc to write to Pegasus. This would be great :) but probably not trivial.

Some notes:
 1. FactoryBot can initialize any object, not just Rails models. Of course it's designed particularly for that and `build` vs. `create` make more sense in that context. We do have [examples](https://github.com/code-dot-org/code-dot-org/blob/89621da91f506c12ca3c7906df297458850cad2d/dashboard/test/factories/pd_factories.rb#L120) of factories that create other classes.
 2. I really like the idea of some kind of wrapper class, model or otherwise. We don't necessarily need to go all the way with creation at first, could start with just queries. We could write unit tests for that model to make sure all the queries work, making the appropriate Pegasus queries. Then we can still use stubs (or better [expectations](http://gofreerange.com/mocha/docs/Mocha/Expectation.html)) here in this test, to ensure the correct "model" query methods are called. This way all the Pegasus calls would be encapsulated inside a well-tested wrapper so we don't need to know about Pegasus in this context. 
 3. Of course the ideal if we have bandwidth for it is a full fledged model wrapper as Dave suggested, or moving the entire table to dashboard and making a proper model.

Reply:
I'm in favor of a wrapper class, but think there's already enough going on in this PR that it seems like a separate undertaking.  I'd also like to do a little more research/see how much time we can commit before implementing a plan.  Thanks for your thoughts and suggestions!